### PR TITLE
tests(audioplayer): add tests

### LINKED ISSUE #29: Add audioplayer tests
There are no tests yet for audio-player.js

### PR DESCRIPTION
**Pull request by Lehren**
Add 100% coverage tests.

Resolve #29

### CODE REVIEW

File: audio-player.js [hunk 1]
```diff
@@ -84,40 +84,40 @@ module.exports = class AudioPlayer {
       throw new Error(
         'Some form of audio playback capability is required');
     }
+    if (this.canUseAudio) {
```
Comment (remcohaszing) [inline audio-player.js]: This check is unnecessary. It can never be true due to the if statement on line 83.

File: test/audio-playerSpec.js [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+    spyOn(AudioPlayer.prototype, '_getBestPlayer');
+  });
+
+  it('should construct with event functionality', () => {
+    const player = new AudioPlayer();
+    player.emitter = {
+      on: jasmine.createSpy(),
+      off: jasmine.createSpy(),
+      emit: jasmine.createSpy()
+    };
```
Comment (remcohaszing) [inline test/audio-playerSpec.js]: You can use this:

```js
player.emitter = jasmine.createSpyObj('emitter', [
  'on',
  'off',
  'emit'
]);
```

File: test/audio-playerSpec.js [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+    window.allOff = jasmine.createSpy();
```
Comment (remcohaszing) [inline test/audio-playerSpec.js]: Why is this assigned to window?

File: test/audio-playerSpec.js [hunk 4]
```diff
@@ -0,0 +1,279 @@
+const AudioPlayer = require('../audio-player');
+const CordovaMediaPlayer = require('../cordova-media-player');
+const WebAudioPlayer = require('../web-audio-player');
+
+describe('Audio player', () => {
+  const oldMedia = window.Media;
+  const oldAudio = window.Audio;
```
Comment (remcohaszing) [inline test/audio-playerSpec.js]: Set this in beforeEach.

File: test/audio-playerSpec.js (hunk 3 above)
Comment (Lehren) [inline test/audio-playerSpec.js]: allOff is function imported from the event-emitter package in audio-player.js

`const allOff = require('event-emitter/all-off');`

Comment (remcohaszing) [inline test/audio-playerSpec.js]: So it's not on window, right?

File: audio-player.js (hunk 1 above)
Comment (Lehren) [inline audio-player.js]: `!(this.canUseAudio || this.canUseCordovaMedia)`

Will only pass if both are false.

`canUseAudio` can still be true, but `canUseCordovaMedia` does not need to. 

It's a bit convoluted though I made it better now.

**Pull request merged**